src/services/BlockchainScanner.ts

```diff
@@ -44,25 +59,53 @@ export abstract class BlockchainScanner {
     return {fromBlock, toBlock, success, synchronized: toBlock < fromBlock + this.step};
   }
 
-  private async getToBlock(fromBlock: number): Promise<number> {
+  setChainId(chainId: ChainsIds) {
+    this.chainId = chainId;
+  }
+
+  setProvider() {
+    if (!this.chainId) {
+      this.logger.error('[UniswapPoolScanner] chainId not set');
```

Comment:
this is not Uniswap scanner, this is block scanner.

I dont think setters will work for injectable classes. it must be repository/factory pattern, we using it almost everywhere.

Reply:
Okay, I will change to the correct pattern